Libraries/NavigationExperimental/NavigationTypeDefinition.js

```diff
@@ -23,7 +23,8 @@ export type NavigationGestureDirection = 'horizontal' | 'vertical';
 
 export type NavigationRoute = {
   key: string,
-  title?: string
+  title?: string,
+  direction?: NavigationGestureDirection,
```

Comment:
Thanks for your contribution.

I'm concerned about that the route is becoming the convenient place that people could add anything to it.
I'd imagine that in few months later, the route will become something like this.

```
export type NavigationRoute = {
  key: string,
  title?: string,
  subTitle?: string,
  direction? (number|string),
  velocity? number,
  id? (number|string),
  blah: any,
  blahBlahBlah: any,
};
```

and so will people keep adding new things to `NavigationState`.

If this pattern is accepted, will this be scalable to support different gestures, animations, rendering needs?  

IMHO, we do allow people to add anything to the route, on the other hand, we don't want to limit the types of data that the route has or can carry.

Maybe we should explicit define a prop `meta` that can be anything thus people can add whatever type of data to it without restricting others. 

```
export type NavigationRoute = {
  key: string,
  meta?: any, 
};
```